services/storage-users/pkg/command/trash_bin.go

```diff
@@ -53,3 +89,402 @@ func PurgeExpiredResources(cfg *config.Config) *cli.Command {
+			case "skip":
+				overwriteOption = SKIP
+			case "replace":
+				overwriteOption = REPLACE
+			case "keep-both":
+				overwriteOption = KEEP_BOTH
+			default:
+				_ = cli.ShowSubcommandHelp(c)
+				return cli.Exit("The option flag is invalid", 1)
+			}
+
+			ref, err := storagespace.ParseReference(spaceID)
+			if err != nil {
+				return err
+			}
+			client, err := pool.GetGatewayServiceClient(cfg.RevaGatewayGRPCAddr)
+			if err != nil {
+				log.Error().Err(err).Msg("error selecting gateway client")
+				return err
+			}
+			ctx, err := utils.GetServiceUserContext(cfg.ServiceAccount.ServiceAccountID, client, cfg.ServiceAccount.ServiceAccountSecret)
+			if err != nil {
+				log.Error().Err(err).Msg("could not impersonate")
+				return err
+			}
+
+			spanOpts := []trace.SpanStartOption{
+				trace.WithSpanKind(trace.SpanKindClient),
+				trace.WithAttributes(
+					attribute.KeyValue{Key: "option", Value: attribute.StringValue(optionFlagVal)},
+					attribute.KeyValue{Key: "spaceID", Value: attribute.StringValue(spaceID)},
+					attribute.KeyValue{Key: "itemID", Value: attribute.StringValue(itemID)},
+				),
+			}
+			ctx, span := tp.Tracer("storage-users trash-bin restore").Start(ctx, "serve static asset", spanOpts...)
+			defer span.End()
+
+			res, err := client.ListRecycle(ctx, &provider.ListRecycleRequest{Ref: &ref, Key: "/"})
+			if err != nil {
+				log.Error().Err(err).Msg(retrievingErrorMsg)
+				return err
+			}
+			if res.Status.Code != rpc.Code_CODE_OK {
+				return fmt.Errorf("%s %s", retrievingErrorMsg, res.Status.Code)
+			}
+
+			var found bool
+			var itemRef *provider.RecycleItem
+			for _, item := range res.GetRecycleItems() {
+				if item.GetKey() == itemID {
+					itemRef = item
+					found = true
+					break
+				}
+			}
+			if !found {
+				return fmt.Errorf("itemID '%s' not found", itemID)
+			}
+			fmt.Printf("\nRun restoring with option=%s\n", optionFlagVal)
+			fmt.Printf("restoring itemID: '%s', path: '%s', type: '%s'\n", itemRef.GetKey(), itemRef.GetRef().GetPath(), itemType(itemRef.GetType()))
+			dstRes, err := restore(ctx, client, ref, itemRef, overwriteOption)
+			if err != nil {
+				return err
+			}
+			fmt.Printf("itemID: '%s', path: '%s', restored as '%s'\n", itemRef.GetKey(), itemRef.GetRef().GetPath(), dstRes.GetPath())
+			return nil
+		},
+	}
+}
+
+func restore(ctx context.Context, client gateway.GatewayAPIClient, ref provider.Reference, item *provider.RecycleItem, overwriteOption int) (*provider.Reference, error) {
+	dst, _ := deepcopy.Copy(ref).(provider.Reference)
+	dst.Path = utils.MakeRelativePath(item.GetRef().GetPath())
+	// Restore request
+	req := &provider.RestoreRecycleItemRequest{
+		Ref:        &ref,
+		Key:        path.Join(item.GetKey(), "/"),
+		RestoreRef: &dst,
+	}
+
+	exists, dstStatRes, err := isDestinationExists(ctx, client, dst)
+	if err != nil {
+		return &dst, err
+	}
+
+	if exists {
+		fmt.Printf("destination '%s' exists.\n", dstStatRes.GetInfo().GetPath())
```

Comment:
use `log.Info` instead `fmt.Printf`

Reply:
I used the fmt because it should always be visible.